### PR TITLE
Actix update

### PR DESCRIPTION
**Pull request by kgrech**
Right now actix-web v3 creates tokio v0.2.25 Runtime. As a result, we need to include 2 versions of tokio and run different versions of runtime in parallel. Once updated, it stats using tokio v 1.7, same as the one used by the rest of the application.

However, v 1.7 enforces a few reasonable checks:
- It is not allowing ``block_on`` calls inside async context
- It does not allow the runtime drop inside the async context (since it does ``block_on``)

To fix the above, the PR refactors a set of API make them async to replace block_on with usual await. It should also have a positive performance impact, as the above awaits are not blocking the reactor thread anymore.

Finally, the PR updates the test suite to leverage ``tokio::test`` to avoid manual runtime creation. The side effect is that the same runtime is used to run the test method and as the runtime for the searcher, which is different from the case in production where the production code runs in actix created runtime and the searcher runs in its own runtime. However, since the searcher runtime is global anyway, this should not affect the quality of regression testing, however, it simplifies a code a lot. Moreover, I assume we might merge searcher and actix runtime into the same one in production at some point as well.

### All Submissions:

* [X] Have you followed the guidelines in our Contributing document?
* [X] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?

### New Feature Submissions:

1. [X] Does your submission pass tests?
2. [X] Have you lint your code locally using ``cargo fmt`` command prior to submission?
3. [X] Have you checked your code using ```cargo clippy``` command?

### CODE REVIEW

File: src/api/recommend_api.rs [hunk 1]
```diff
@@ -2,24 +2,31 @@ use crate::common::helpers::process_response;
 use actix_web::rt::time::Instant;
 use actix_web::{post, web, Responder};
 use collection::operations::types::RecommendRequest;
+use segment::types::ScoredPoint;
 use std::sync::Arc;
+use storage::content_manager::errors::StorageError;
 use storage::content_manager::toc::TableOfContent;
 
+async fn do_recommend_points(
```
Comment (kgrech) [inline src/api/recommend_api.rs:10]: Here and below had to move it out to it's own method as I can't call async method inside ``and_then`` - it accepts sync closure. But I would need it to be a separate method to be called from the GRPC anyway. I'll move them out into ``common`` module and remove do_ in the next PR where I expose actix as a feature and add grpc

File: src/main.rs [hunk 2]
```diff
@@ -64,11 +65,11 @@ async fn main() -> std::io::Result<()> {
         App::new()
             .wrap(Logger::default())
             .app_data(toc_data.clone())
-            .data(
+            .app_data(Data::new(
```
Comment (kgrech) [inline src/main.rs:68]: ``data`` was deprecated

File: src/api/search_api.rs [hunk 3]
```diff
@@ -2,24 +2,32 @@ use crate::common::helpers::process_response;
...
+    toc.get_collection(&name)?
+        .search(Arc::new(request))
+        .await
+        .map_err(|err| err.into())
+}
+
 #[post("/collections/{name}/points/search")]
 pub async fn search_points(
     toc: web::Data<TableOfContent>,
-    web::Path(name): web::Path<String>,
```
Comment (kgrech) [inline src/api/search_api.rs:11]: In a new actix you can't do ``web::Path(name)`` anymore, you have to use ``into_inner``.

File: lib/collection/src/collection.rs [hunk 4]
```diff
@@ -308,6 +325,16 @@ impl Collection {
...
+
+        // The drop could be called from the tokio context, e.g. from perform_collection_operation method.
+        // Calling remove from there would lead to the following error in a new version of tokio:
+        // "Cannot drop a runtime in a context where blocking is not allowed. This happens when a runtime is dropped from within an asynchronous context."
+        // So the workaround for move out the runtime handler and drop it in the separate thread.
+        // The proper solution is to reconsider the collection to be an owner of the runtime
+        let handle = self.runtime_handle.take();
+        thread::spawn(move || {
+            drop(handle);
+        });
```
Comment (generall) [inline lib/collection/src/collection.rs:338]: Since the `runtime_handle` is shared between collections, why should we drop it on collection destruction?

Comment (kgrech) [reply lib/collection/src/collection.rs:338]: It is not shared. It is actually an optimizer runtime handler which is collection specific

Comment (generall) [reply lib/collection/src/collection.rs:338]: oh, yes, you are right.

**Pull request merged**